traj/src/traj/synchronize_joint_motion.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python
+import math
+import numpy as np
+import traj 
+import rospy
+
+
+def synchronize_joint_motion( t_syn, pos_diff, v_start, v_end, abs_max_pos, abs_max_vel, abs_max_acc, abs_max_jrk):
+	abs_v_start = abs(v_start)
+	abs_v_end = abs(v_end)
+	jm = abs_max_jrk
+	## step 1: calculate all variables that determine which equation will be used for synchronization
+	tj_2vf, ta_2vf, tj, ta, tv = traj.traj_segment_planning(0.0, pos_diff, abs_v_start, abs_v_end, abs_max_vel, abs_max_acc, abs_max_jrk)
+	minPos_2vf, acc_2vf, tj_2vf, ta_2vf = traj.calculate_minPos_reachAcc_maxJrkTime_maxAccTime_to_final_vel( v_start, v_end, abs_max_vel, abs_max_acc, abs_max_jrk)
```

Comment:
`calculate_minPos_reachAcc_maxJrkTime_maxAccTime_to_final_vel(..)` will probably not make @jonbinney very happy, but I'd be interested in an alternative name he can come up with ;)

Reply:
haha it is a descriptive name, which is good! Maybe we just standardize the case and use of underscores? ` `min_pos_reach_acc_max_jerk_time_max_acc_time_to_final_velocity()`?

 Looks like that function is defined here: https://github.com/iron-ox/trajectory_smoothing/blob/fef250a11d0e4d08cad1b3360adfce38cb5dd946/traj/src/traj/segment_planning.py#L165

From reading the comment before the function definition, i'm not sure what the `maxJrkTime` and `maxAccTime` portions  mean. For functions like that which do something very specific, it would be great to have a docstring that explains each part of the name ot the reader.

Reply:
Actually i take that back - if the name is taken character-for-character from a paper, and the function is intended only for use within the file (within the context of the implementation of that  algorithm from that research paper) then I think it is reasonable to ignore the typical python naming style. It would be good to have a docstring for each function though, letting the reader know exactly where to look. e.g.
```
This function is an implementation of .... from the pseudocode in table X of [1].

<high level description of what the function does>
```